drltrace_src/drltrace.cpp

```diff
@@ -80,85 +80,114 @@ static std::vector<std::string> filter_module_blacklist;
-    case DRSYS_TYPE_SIGNED_INT:   print_simple_value(arg, false); break;
-    case DRSYS_TYPE_UNSIGNED_INT: print_simple_value(arg, false); break;
-    case DRSYS_TYPE_HANDLE:       print_simple_value(arg, false); break;
-    case DRSYS_TYPE_NTSTATUS:     print_simple_value(arg, false); break;
-    case DRSYS_TYPE_ATOM:         print_simple_value(arg, false); break;
-#ifdef WINDOWS
-    case DRSYS_TYPE_LCID:         print_simple_value(arg, false); break;
-    case DRSYS_TYPE_LPARAM:       print_simple_value(arg, false); break;
-    case DRSYS_TYPE_SIZE_T:       print_simple_value(arg, false); break;
-    case DRSYS_TYPE_HMODULE:      print_simple_value(arg, false); break;
-#endif
+    case DRSYS_TYPE_VOID:         print_simple_value(arg, true, output); break;
+    case DRSYS_TYPE_POINTER:      print_simple_value(arg, true, output); break;
+    case DRSYS_TYPE_BOOL:         print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_INT:          print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_SIGNED_INT:   print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_UNSIGNED_INT: print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_HANDLE:       print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_NTSTATUS:     print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_ATOM:         print_simple_value(arg, false, output); break;
+        #ifdef WINDOWS
+    case DRSYS_TYPE_LCID:         print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_LPARAM:       print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_SIZE_T:       print_simple_value(arg, false, output); break;
+    case DRSYS_TYPE_HMODULE:      print_simple_value(arg, false, output); break;
+        #endif
     case DRSYS_TYPE_CSTRING:
-        print_string(drcontext, (void *)arg->value, false);
+        print_string(drcontext, (void *)arg->value, false, output);
         break;
     case DRSYS_TYPE_CWSTRING:
-        print_string(drcontext, (void *)arg->value, true);
+        print_string(drcontext, (void *)arg->value, true, output);
         break;
     default: {
         if (arg->value == 0)
-            dr_fprintf(outf, "<null>");
-        else
-            dr_fprintf(outf, PFX, arg->value);
+            output.append("<null>");
+        else {
+            char str[128];
+            sprintf(str, PFX, arg->value);
+            output.append("0x");
+            output.append(str);
+        }
     }
     }
-
-    dr_fprintf(outf, " (%s%s%stype=%s%s, size=" PIFX ")",
-              (arg->arg_name == NULL) ? "" : "name=",
-              (arg->arg_name == NULL) ? "" : arg->arg_name,
-              (arg->arg_name == NULL) ? "" : ", ",
-              (arg->type_name == NULL) ? "\"\"" : arg->type_name,
-              (arg->type_name == NULL ||
-              TESTANY(DRSYS_PARAM_INLINED|DRSYS_PARAM_RETVAL, arg->mode)) ? "" : "*",
-              arg->size);
+    char str[128];
```

Comment:
same here

Reply:
and in a lot of places below.

Reply:
As I fill it with a string representing a pointer address, I thought that 128 should be more than enough to contain it.
To fill it I could use snprintf instead of sprintf to avoid overflow?